rotkehlchen/chain/aggregator.py

```diff
@@ -1292,3 +1293,38 @@ def get_chain_manager(
         """Returns blockchain manager"""
         attr = blockchain.name.lower()
         return getattr(self, attr)
+
+    def add_accounts_to_all_evm(
+            self,
+            accounts: list[ChecksumEvmAddress],
+    ) -> list[tuple[SUPPORTED_EVM_CHAINS, ChecksumEvmAddress]]:
+        """Adds each account for all evm addresses
+
+        Counting ethereum mainnet as the main chain we check if the account is a contract
+        in mainnet. If not we add it for all other chains. If yes we just keep mainnet.
+        If it's already added in a chain we just ignore that chain.
+
+        Returns a list of tuples of the address and the chain it was added in
+        """
+        added_accounts = []
+        for account in accounts:
+            chains: list[SUPPORTED_EVM_CHAINS]
+            if self.ethereum.node_inquirer.get_code(account) != '0x':
+                log.debug(f'Not adding {account} to all evm chains as it is a mainnet contract')
+                chains = [SupportedBlockchain.ETHEREUM]
```

Comment:
The case of a contract in other chains add extra complexity and slows the addition. Is that why you didn't consider it?

Reply:
No no. If it's a contract then you don't own the same address in any other chain. EOA you own the private key and it's repeatable in all chains. Contract account no. So for a contract account we should not do multi-evm addition.